lib/dogma.ex

```diff
@@ -10,13 +10,33 @@ defmodule Dogma do
   alias Dogma.Rules
   alias Dogma.ScriptSources
 
-  def run(dir_path \\ "") do
-    dir_path
+  @formatters %{
+    "simple" => Formatter.Simple,
+    "emacs"  => Formatter.Emacs
+  }
+
+  def run(argv) do
+    argv
+    |> parse_args
+    |> process
+  end
+
+  def parse_args(argv) do
```

Comment:
This doesn't quite feel like the right place for this to me, but I'm happy to leave it here for now as I want to overhaul this section of the application in the near future.